src/types/arrow.rs

```diff
@@ -0,0 +1,288 @@
+// Rust Simplicity Library
+// Written in 2023 by
+//   Andrew Poelstra <apoelstra@blockstream.com>
+//
+// To the extent possible under law, the author(s) have dedicated all
+// copyright and related and neighboring rights to this software to
+// the public domain worldwide. This software is distributed without
+// any warranty.
+//
+// You should have received a copy of the CC0 Public Domain Dedication
+// along with this software.
+// If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+//
+
+//! Types Arrows
+//!
+//! Every Simplicity expression has two types associated with it: a source and
+//! a target. See the `types` module for more information.
+//!
+//! This module defines the specific arrows associated with each kind of node.
+
+use std::fmt;
+use std::sync::Arc;
+
+use crate::core::Value;
+use crate::types::{Bound, Error, Final, Type};
+use crate::{jet::Jet, Context};
+
+/// A structure representing an expression's type, which maps a source type to a target type
```

Comment:
"Type of an expression, which maps a source type to a target type"

This is slightly confusing for the user because `Type`s don't contain arrows. These kinds of types here are a generalization. We should explain this in the module docs.